### PR TITLE
Refactor page template

### PR DESCRIPTION
**Pull request by almeynman**
This is my attempt to simplify routing and abstractions in `page` [folder.](url)

What I did:
- Added BlurProvider - it is now only responsible for making the whole app blur and unblur;
- Added ProgressProvider - it is now only responsible for showing/not showing the green progress line at the top of the page
- Added MobileAppFrame - it is a component responsible for making web version look like mobile app
- Added RouteWithLayout - it's responsibility is to wrap a routed component in a layout
- Added BottomNavLayout - an example of how we can use layouts. Probably will change
- Removed appWrapper, authenticatedPage, verifiedPage, as they did more then one thing and when combined did the same thing.

### CODE REVIEW

**Comment by jayrmotta** [comment]
Hey @almeynman your PR seems to add some rad stuff, can you put up some more detailed description on the reasons and purposes of the components / code you've introduced?

I think this can be beneficial to help everyone understand and reuse everyone's code, also this create a better PR culture.

Cheers bro! :)

File: src/page/progressProvider/ProgressProvider.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// @flow
+
+import * as React from 'react';
+import { connect } from 'react-redux';
+import type { MapStateToProps } from 'react-redux';
+import { withRouter } from 'react-router';
+import { Loader } from 'ui';
+
+type Props = {
+  progress: boolean,
```
Comment (jayrmotta) [inline src/page/progressProvider/ProgressProvider.js]: The idea is that if I change the progress application state then a loader will be rendered?

File: src/page/blurProvider/BlurProvider.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+import * as React from 'react';
+import { connect, type MapStateToProps } from 'react-redux';
+import { keyframes } from 'styled-components';
+import { CSSTransition } from 'react-transition-group';
+import { withRouter } from 'react-router';
+import { Content } from 'ui';
+
+type Props = {
+  blur: boolean,
```
Comment (jayrmotta) [inline src/page/blurProvider/BlurProvider.js]: Is this intended to render something on top of everything else? Or just blur so the use won't take any action during some other thing?

Comment (almeynman) [reply src/page/blurProvider/BlurProvider.js]: The only place I have seen this used is in the modal component to blur the background. Reducer and actions for blur are currently located in `page/` folder. I think you can safely use it to blur the page for whatever use case

File: src/page/progressProvider/ProgressProvider.js (hunk 1 above)
Comment (almeynman) [reply src/page/progressProvider/ProgressProvider.js]: Yes, this guy is responsible for showing the green line at the top of the screen, feel free to use it. Actions and reducers for this component are currently located in `page/` folder

**Pull request merged**